app/src/main/java/com/example/intervaltimer/TimerActivity.kt

```diff
@@ -73,39 +94,75 @@ class TimerActivity : AppCompatActivity() {
         }
 
         fastForwardButton.setOnClickListener{
-            mediaPlayer?.stop()
-            totaltime = 70000 + 1000
-            workoutName.text = "Curls" // placeholder code for setting text when hitting forward
-            repsText.text = "# of reps 80" // placeholder code for setting text when hitting forward
-            timerText.text = "01:10" // placeholder code for setting text at startup
-            playButton.setBackgroundResource(R.drawable.ic_play)
-            pause()
-            buttonSwitch = false
+            forwardBackword(1)
         }
 
         rewindButton.setOnClickListener{
-            mediaPlayer?.stop()
-            totaltime = 10000 + 1000
-            workoutName.text = "Push ups" // placeholder code for setting text when hitting back
-            repsText.text = "# of reps 50" // placeholder code for setting text when hitting back
-            timerText.text = "00:10" // placeholder code for setting text at startup
-            playButton.setBackgroundResource(R.drawable.ic_play)
-            pause()
-            buttonSwitch = false
+            forwardBackword(-1)
         }
 
         endWorkoutButton.setOnClickListener{
             mediaPlayer?.release()
             mediaPlayer = null
+            finish()
         }
 
 
         // val audioManager: AudioManager = getSystemService(AUDIO_SERVICE) as AudioManager // Audio manager in case we need it
 
     }
 
+    private fun forwardBackword(nav: Int){
+
+        if (itera == 0 && nav == -1){
+
+        } else if(itera == (args.intervalList.size - 1)){
+
+        } else{
+            itera += nav
+        }
+
+        workoutName.text = args.intervalList[itera].name // placeholder code for setting text at startup
+
+        if (args.intervalList[0].reps.toString() != "null"){
+            repsText.text = args.intervalList[itera].reps.toString() // placeholder code for setting text at startup
+        }
+
+        val covertTime: Int? = args.intervalList[itera].time
```

Comment:
Lines 125-146 are repeated elsewhere in this class, pull them out into a function